src/compiler/scala/tools/nsc/backend/jvm/opt/Inliner.scala

```diff
@@ -440,6 +440,19 @@ abstract class Inliner {
     // label for the exit of the inlined functions. xRETURNs are replaced by GOTOs to this label.
     val postCallLabel = newLabelNode
     clonedInstructions.add(postCallLabel)
+    if (sameSourceFile) {
+      BytecodeUtils.previousLineNumber(callsiteInstruction) match {
+        case Some(line) =>
+          BytecodeUtils.nextExecutableInstruction(callsiteInstruction).flatMap(BytecodeUtils.previousLineNumber) match {
+            case Some(line1) =>
+              if (line == line1)
```

Comment:
We could also just add the line number unconditionally and rely on `removeEmptyLineNumbers` to remove it if it's redundant. The inliner creates other redundant code, especially local variables that are then removed by copy propagation.

Either way is fine.